### ISSUE TITLE
[Possible Bug] Invidious suddenly stops working and uses 100% CPU

### ISSUE THREAD

**Issue by D3V1LC0D3R** [iv-org/invidious#2413]
**Describe the bug**
When normally using invidious, invidious suddenly stops working sometimes and always times-out afterwards.

**Logs**
```
2021-09-19 01:36:10 UTC [trace] fetch_channel: UCpK5XWtxTuCtsKkwQnJbNIw : video dn4B-DJfRFc : Updated
2021-09-19 01:36:10 UTC [trace] RefreshChannelsJob: UCpK5XWtxTuCtsKkwQnJbNIw fiber : Updating DB
2021-09-19 01:36:10 UTC [debug] RefreshChannelsJob: UCpK5XWtxTuCtsKkwQnJbNIw fiber : Done
2021-09-19 01:36:10 UTC [trace] RefreshChannelsJob: Fiber limit ok, continuing
2021-09-19 01:36:10 UTC [debug] RefreshChannelsJob: UCdY369z7KTrdB3hjr_YwPuw : Spawning fiber
2021-09-19 01:36:10 UTC [trace] RefreshChannelsJob: Fiber limit reached, waiting...
2021-09-19 01:36:10 UTC [trace] RefreshChannelsJob: UCdY369z7KTrdB3hjr_YwPuw fiber : Fetching channel
2021-09-19 01:36:10 UTC [debug] fetch_channel: UCdY369z7KTrdB3hjr_YwPuw
2021-09-19 01:36:10 UTC [trace] fetch_channel: UCdY369z7KTrdB3hjr_YwPuw : pull_all_videos = false, locale = 
2021-09-19 01:36:10 UTC [trace] fetch_channel: UCdY369z7KTrdB3hjr_YwPuw : Downloading RSS feed
```
after this the logs stops even i just fetched the log (and tried to open multiple videos

**Additional context**
Linux [HOSTNAME] 3.10.0-1062.12.1.vz7.131.10 #1 SMP Mon Mar 16 15:39:59 MSK 2020 x86_64 GNU/Linux
2 CPUs

config.yml
```
---
channel_threads: 4
feed_threads: 4
output: invidious.log
log_level: All
db:
  user: kemal
  password: [PASSWORD]
  host: localhost
  port: 5432
  dbname: invidious
database_url: postgres://kemal:[PASSWORD]@localhost:5432/invidious
decrypt_polling: true
full_refresh: false
https_only: false
use_pubsub_feeds: false
popular_enabled: false
captcha_enabled: true
login_enabled: true
registration_enabled: false
statistics_enabled: false
admins:
- admin
default_user_preferences:
  annotations: false
  annotations_subscribed: false
  autoplay: false
  captions:
  - ""
  - ""
  - ""
  comments:
  - youtube
  - ""
  continue: false
  continue_autoplay: true
  dark_mode: ""
  latest_only: false
  listen: false
  local: false
  locale: en-US
  max_results: 40
  notifications_only: false
  player_style: invidious
  quality: hd720
  quality_dash: auto
  default_home: Popular
  feed_menu:
  - Popular
  - Trending
  - Subscriptions
  - Playlists
  related_videos: true
  sort: published
  speed: 1.0
  thin_mode: false
  unseen_only: false
  video_loop: false
  extend_desc: false
  volume: 100
  vr_mode: true
dmca_content: []
check_tables: false
cache_annotations: false
hsts: true
disable_proxy: false
force_resolve: 
port: 3000
host_binding: 0.0.0.0
pool_size: 1000
use_quic: true
cookies: ""
captcha_api_url: https://api.anti-captcha.com

```

**Comment by D3V1LC0D3R** [comment]
Its also strange that even with logmode "All" the logs suddenly stop even though it normaly spams the logs with traces

**Comment by unixfox** [comment]
Please post your invidious version, you can find it at the bottom of every invidious web page.

**Comment by D3V1LC0D3R** [comment]
2021.09.16-e655af2

**Comment by unixfox** [comment]
Can you give to us the result of this command:
```
curl https://www.youtube.com/feeds/videos.xml?channel_id=UCeeFfhMcJa1kjtfZAGskOCA
``` ?

We want to know if your server is blocked by Google or not.

**Comment by D3V1LC0D3R** [comment]
doesnt seem so. It gives me a xml
command: ```curl "https://www.youtube.com/feeds/videos.xml?channel_id=UCeeFfhMcJa1kjtfZAGskOCA" |head -n 20```
returns:
```<feed xmlns:yt="http://www.youtube.com/xml/schemas/2015" xmlns:media="http://search.yahoo.com/mrss/" xmlns="http://www.w3.org/2005/Atom">
 <link rel="self" href="http://www.youtube.com/feeds/videos.xml?channel_id=UCeeFfhMcJa1kjtfZAGskOCA"/>
 <id>yt:channel:UCeeFfhMcJa1kjtfZAGskOCA</id>
 <yt:channelId>UCeeFfhMcJa1kjtfZAGskOCA</yt:channelId>
 <title>TechLinked</title>
 <link rel="alternate" href="https://www.youtube.com/channel/UCeeFfhMcJa1kjtfZAGskOCA"/>
 <author>
  <name>TechLinked</name>
  <uri>https://www.youtube.com/channel/UCeeFfhMcJa1kjtfZAGskOCA</uri>
 </author>
 <published>2018-05-03T19:49:25+00:00</published>
 <entry>
  <id>yt:video:CWHhO8NAJOU</id>
  <yt:videoId>CWHhO8NAJOU</yt:videoId>
  <yt:channelId>UCeeFfhMcJa1kjtfZAGskOCA</yt:channelId>
  <title>The Pixel 6 could be FIRE</title>
  <link rel="alternate" href="https://www.youtube.com/watch?v=CWHhO8NAJOU"/>
  <author>
   <name>TechLinked</name>```

**Comment by unixfox** [comment]
Duplicate of https://github.com/iv-org/invidious/issues/1438 and https://github.com/iv-org/invidious/issues/1439.

Invidious has some memory issues which are related the issues that you are encountering.

We recommend you to automatically restart Invidious every hour in order to "temporarily" fix the issue.
